dmscripts/notify_buyers_to_award_closed_briefs.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, date, timedelta
+
+import dmapiclient
+from dmutils.email.dm_notify import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmutils.formats import DATE_FORMAT
+
+from dmscripts.helpers import logging_helpers, date_helpers, brief_data_helpers
+from dmscripts.helpers.logging_helpers import logging
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging.INFO})
+
+
+def _filter_brief_users_by_optional_user_id_list(brief, user_id_list):
+    """Filter the brief's users by restricted user ID list if present
+    """
+    if user_id_list:
+        return filter(lambda user: user['id'] in user_id_list, brief['users'])
+    else:
+        return brief['users']
+
+
+def _filter_briefs_by_optional_user_id_list(closed_briefs, user_id_list):
+    """Return briefs if they have any users
+    These users can be optionally filtered by the user_id_list
+    """
+    return [
+        brief for brief in closed_briefs if any(
+            _filter_brief_users_by_optional_user_id_list(brief, user_id_list)
+        )
+    ]
+
+
+def _create_context_for_brief(brief):
+    return {
+        'brief_id': brief['id'],
+        'brief_title': brief['title'],
+        'utm_date': date.today().strftime("%Y%m%d")
+    }
+
+
+def notify_users(notify_client, notify_template_id, brief, user_id_list):
+    failed_users = []
+    if brief['users']:
+        context_data = _create_context_for_brief(brief)
+        brief_users = _filter_brief_users_by_optional_user_id_list(brief, user_id_list)
+
+        for user in brief_users:
+            if user['active']:
+                try:
+                    notify_client.send_email(
+                        user['emailAddress'], notify_template_id, context_data, allow_resend=False
+                    )
+                except EmailError:
+                    failed_users.append(user['id'])
+
+    return failed_users
+
+
+def _check_that_brief_closing_date_before_the_offset(date_closed, offset_days):
+    if date_closed:
```

Comment:
I found this a bit tough to get my head around. I think because `date_closed` is used a lot in the function as well as `main`. Might be worth adding a comment?